src/services/User.tsx

```diff
@@ -39,14 +45,55 @@ export const getDrafts = async (user: string) => {
   return { drafts }
 }
 
+
+// export const getUserNFTs = async (user: string) => {
+// const userNFTs = await getOwnerNFTs(user)
+// console.log('user NFT', userNFTs)
+//  const onMarket: NFT[] = []
+//   const offMarket: NFT[] = []
+
+// for (var i in userNFTs) {
+// console.log('i=>', i)
+// const auctions = await getNFTAuctions(user)
+// console.log('auctions=>', auctions)
+// if (auctions.filter(auction => auction.status === 'open').length > 0) onMarket = onMarket.concat(userNFTs[i])
+// else offMarket.push(userNFTs[i])
+// console.log('onMarket=>', onMarket)
+// }
+
+// return { onMarket, offMarket }
+// }
+
+// get onMarket offMarket NFT
 export const getUserNFTs = async (user: string) => {
   const userNFTs = await getOwnerNFTs(user)
-  const onMarket: NFT[] = []
-  const offMarket: NFT[] = []
-  for (var i in userNFTs) {
-    const auctions = await getNFTAuctions(userNFTs[i].id)
-    if (auctions.filter(auction => auction.status === 'open').length > 0) onMarket.push(userNFTs[i])
-    else offMarket.push(userNFTs[i])
-  }
+  const NftArray: NFT[] = [...Object.values(userNFTs)]
+  const auctions: any = []
```

Comment:
why didn't you use the hooks I already created they are much simpler @abdelaziz013

Reply:
it is not working

Reply:
we could debug them but the Flow of the logic can be the same